packages/core/contracts/financial-templates/perpetual-multiparty/ConfigStore.sol

```diff
@@ -0,0 +1,148 @@
+     * @return ConfigSettings config settings that calling financial contract should view as "live".
+     */
+    function getCurrentConfig() external view override nonReentrantView() returns (ConfigSettings memory) {
+        if (_pendingProposalPassed()) {
+            return pendingConfig;
+        } else {
+            return currentConfig;
+        }
+    }
+
+    /**
+     * @notice Propose new configuration settings. New settings go into effect
+     * after a liveness period passes.
+     * @dev Callable only by owner. Calling this while there is already a pending proposal
+     * will overwrite the pending proposal.
+     */
+    function proposeNewConfig(ConfigSettings memory newConfig) external onlyOwner() nonReentrant() updateConfig() {
+        // Alternatively, we can revert if there is already a pending proposal.
+        // require(pendingPassedTimestamp == 0, "Pending proposal exists");
+
+        _validateConfig(newConfig);
+
+        // Warning: This overwrites a pending proposal!
+        pendingConfig = newConfig;
+
+        // Use current config's liveness period to timelock this proposal.
+        pendingPassedTimestamp = getCurrentTime() + currentConfig.updateLiveness;
+
+        emit ProposedNewConfigSettings(
+            msg.sender,
+            newConfig.rewardRatePerSecond.rawValue,
+            newConfig.proposerBondPct.rawValue,
+            newConfig.updateLiveness,
+            pendingPassedTimestamp
+        );
+    }
+
+    function publishPendingConfig() external nonReentrant() updateConfig() {}
+
+    /****************************************
+     *         INTERNAL FUNCTIONS           *
+     ****************************************/
+
+    // Check if pending proposal can overwrite the current config.
+    function _updateConfig() internal {
+        // If liveness has passed, publish new reward rate.
+        if (_pendingProposalPassed()) {
+            currentConfig = pendingConfig;
+
+            _deletePendingConfig();
+
+            emit ChangedNewConfigSettings(
+                currentConfig.rewardRatePerSecond.rawValue,
+                currentConfig.proposerBondPct.rawValue,
+                currentConfig.updateLiveness
+            );
+        }
+    }
+
+    function _deletePendingConfig() internal {
+        delete pendingConfig;
+        pendingPassedTimestamp = 0;
+    }
+
+    function _pendingProposalPassed() internal view returns (bool) {
+        return (pendingPassedTimestamp != 0 && pendingPassedTimestamp <= getCurrentTime());
+    }
+
+    // Use this method to constrain values with which you can set ConfigSettings.
+    function _validateConfig(ConfigSettings memory config) internal pure {
+        // Make sure updateLiveness is not too long, otherwise contract can might not be able to fix itself
+        // before a vulnerability drains its collateral.
+        require(config.updateLiveness <= 604800, "Invalid paramUpdateLiveness");
```

Comment:
nit: maybe add a variable or a comment noting how much time this is?

Reply:
ah right I meant to. 7 days